realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -168,6 +168,16 @@ public boolean isClosed() {
         return nativePtr == 0 || nativeIsClosed(nativePtr);
     }
 
+    public void writeCopy(File file, byte[] key) {
+        if (file.isFile() && file.exists()) {
+            throw new IllegalArgumentException("The destination file must not exist");
+        }
+        if (key != null && key.length != 64) {
```

Comment:
Then we have to do the check. But I think it a bit strange that the object store doesn't check.